File: packages/icoder-python/icoder_sdk/resources/reviews.py
"""Reviews (medical coding) resource."""

from __future__ import annotations
from ..client import iCoDerClient
from ..types import Review


class ReviewsResource:
    def __init__(self, client: iCoDerClient):
        self._client = client

    def create(self, encounter_id: str, coding_systems: list[str] | None = None,
               async_mode: bool = False) -> Review:
        params = {"async": "true"} if async_mode else {}
        resp = self._client.post("/api/reviews", json={
            "encounter_id": encounter_id,
            "coding_systems": coding_systems,
        }, params=params)
        resp.raise_for_status()
        data = resp.json()
        return Review(**{k: v for k, v in data.items() if k in Review.__dataclass_fields__})

    def get(self, review_id: str) -> dict:
        resp = self._client.get(f"/api/reviews/{review_id}")
        resp.raise_for_status()
        return resp.json()

    def list(self, page: int = 1, page_size: int = 20) -> dict:
        resp = self._client.get("/api/reviews", params={"page": page, "page_size": page_size})
        resp.raise_for_status()
        return resp.json()

    def review_candidate(self, review_id: str, candidate_id: str, decision: str,
                         reason: str, modified_code: str = "", modified_name: str = "") -> dict:
        resp = self._client.put(f"/api/reviews/{review_id}/candidates/{candidate_id}/review", json={
            "candidate_id": candidate_id, "decision": decision, "reason": reason,
            "modified_code": modified_code, "modified_name": modified_name,
        })
        resp.raise_for_status()
        return resp.json()

    def complete(self, review_id: str, notes: str = "") -> dict:
        resp = self._client.put(f"/api/reviews/{review_id}/complete", json={
            "reviewer_notes": notes, "human_review_status": "completed",
        })
        resp.raise_for_status()
        return resp.json()

File: packages/icoder-sdk/src/resources/reviews.ts
// Reviews resource — medical coding pipeline
import type { AxiosInstance } from 'axios';
import type { Review } from '../types.js';

export class ReviewsResource {
  constructor(private http: AxiosInstance) {}

  async create(encounterId: string, options?: { coding_systems?: string[]; async?: boolean }): Promise<Review> {
    const { data } = await this.http.post<Review>('/api/reviews', {
      encounter_id: encounterId,
      coding_systems: options?.coding_systems,
    }, { params: options?.async ? { async: true } : {} });
    return data;
  }

  async get(id: string): Promise<Review> {
    const { data } = await this.http.get<Review>(`/api/reviews/${id}`);
    return data;
  }

  async list(page = 1, pageSize = 20): Promise<{ items: Review[]; total: number }> {
    const { data } = await this.http.get('/api/reviews', { params: { page, page_size: pageSize } });
    return data;
  }

  async reviewCandidate(reviewId: string, candidateId: string, decision: { decision: string; reason: string; modified_code?: string; modified_name?: string }): Promise<any> {
    const { data } = await this.http.put(`/api/reviews/${reviewId}/candidates/${candidateId}/review`, {
      candidate_id: candidateId,
      ...decision,
    });
    return data;
  }

  async complete(id: string, notes?: string): Promise<Review> {
    const { data } = await this.http.put(`/api/reviews/${id}/complete`, {
      reviewer_notes: notes,
      human_review_status: 'completed',
    });
    return data;
  }

  async getReport(id: string, format: 'markdown' | 'html' = 'markdown'): Promise<string> {
    const { data } = await this.http.get(`/api/reviews/${id}/report/${format}`);
    return data;
  }
}
